### DISCUSSION TITLE
How to center Markdown tables

### DISCUSSION THREAD

**Discussion by JulianOestreich90** [mmistakes/minimal-mistakes#3847] [Q&A]
I have a markdown table which works perfectly fine. As soon as i wrap it with `<div align="center"></div>` or with `<center></center>`. The generated html is bullshit. I'm using a `single` Page Layout and use a table of contents for that blog post.
```
# Total cost

<!-- <div align="center"> -->

|           | **Country** | **Visa Cost**                                        | **Additional Costs**                     |
|-----------|-------------|------------------------------------------------------|------------------------------------------|
|           | Iran        | 80€ Consulate Fee <br>(in Erzurum)                   | ~30€ for Grant Notice<br>from TAP Persia |
|           | Pakistan    | 50€ Fee for Online Visa                              | --                                       |
|           | India       | 96€ Consulate Fee<br>+ Covid Fee (Applied in Berlin) | 24.99€ for IVS Service                   |
| **Total** |             |                                                      | ~281€                                    |

<!-- </div> -->

```
With the `<div>` or `<center>` tags i get this:
![image](https://user-images.githubusercontent.com/19915127/183780368-385669ec-ffb8-4e6f-b146-f7ee2fea662b.png)

**Comment by ryan-p-randall** [comment]
I think there are two different issues here, neither specific to Minimal Mistakes. 

But they shouldn't be hard to fix!

## First: Why isn't this markdown code being turned into an html table?

It looks like this table is being output exactly as it is written in the source text—which is the expected behavior whenever someone puts markdown inside of a span of HTML.

If you're using kramdown (the [default for GitHub Pages](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll)) to render markdown, you can add `markdown="1"` to any opening HTML tag, and then kramdown will know that you want it to interpret any markdown found inside of that [HTML span](https://kramdown.gettalong.org/syntax.html#html-spans).

## Second: Why isn't this content being centered?

[Align is deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#deprecated_attributes), so your browser might just be ignoring it.  

With modern browsers, try using [`style="display: flex; justify-content: center;"`](https://developer.mozilla.org/en-US/docs/Web/CSS/margin#horizontal_centering) instead.

## Okay, so how do these combine?

Putting these together for this particular case, you'll probably want the opening tag to read `<div style="display: flex; justify-content: center;" markdown="1">`.

I hope that helps!
